### PR TITLE
Fix 2 memory leaks

### PR DESCRIPTION
**Pull request by alejandro-colomar**
Hi @ikerexxe ,

This is my implementation with a goto, plus additional refactors to address your concerns about goto, and also simplify some weirdness.

### CODE REVIEW

File: src/usermod.c [hunk 1]
```diff
@@ -685,263 +687,277 @@ fail_exit (int code)
...
+	struct group  *ngrp;
 
-		ngrp = __gr_dup (grp);
-		if (NULL == ngrp) {
-			fprintf (stderr,
-			         _("%s: Out of memory. Cannot update %s.\n"),
-			         Prog, gr_dbname ());
-			fail_exit (E_GRP_UPDATE);
-		}
+	changed = false;
```
Comment (ikerexxe) [inline src/usermod.c:712]: You can set the initial value  when defining the variable

File: src/usermod.c [hunk 2]
```diff
@@ -685,263 +687,277 @@ fail_exit (int code)
...
+	bool         was_admin;
+	struct sgrp  *nsgrp;
 
-		/*
-		 * See if the user specified this group as one of their
-		 * concurrent groups.
-		 */
-		is_member = Gflg && (   (was_member && aflg)
-		                     || is_on_list (user_groups, sgrp->sg_name));
+	changed = false;
```
Comment (ikerexxe) [inline src/usermod.c:834]: You can set the initial value  when defining the variable

**Review by ikerexxe — COMMENTED** [review]
Very good improvement. Thank you man!

I added two inline comments, and I'd also like you to change two commits messages: 14aa5a8ab70326702f22c99c5292f9d40b19c272 and 1ea4f91b6cc444614ef77549ce5bf5d1c5246902. The code explanation is too much, I think it's enough with the first two lines.

File: src/usermod.c (hunk 1 above)
Comment (alejandro-colomar) [reply src/usermod.c:712]: I consciously avoid that.  I find it more readable if the declarations don't initialize, and in general, a safer practice (not in this case, though).

I prefer to have a strong separation between declarations and statements.

File: src/usermod.c (hunk 2 above)
Comment (alejandro-colomar) [reply src/usermod.c:834]: Same as the other case.

**Review by ikerexxe — APPROVED** [review]
I don't see anything blocking from accepting these changes, so I proceed to merge them.

**Pull request merged**